xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -300,9 +300,10 @@ bool CGUIWindowPVRBase::InitChannelGroup()
     if (m_channelGroup != group)
     {
       m_viewControl.SetSelectedItem(0);
-      m_channelGroup = group;
-      m_vecItems->SetPath(GetDirectoryPath());
+      SetChannelGroup(group, false);
```

Comment:
Can't see the full context on the phone so are you sure that both this method and SetChannelGroup() don't attempt to grab the same lock?